### ISSUE TITLE
Trouble parsing xml with XmlSimple

### ISSUE THREAD

**Issue by eperry** [logstash-plugins/logstash-filter-xml#15]
I have an application with dumps a line in XML (Of couse it is multiline and merged with the codec multiline)

I have attached an example error and example XML which I have checked it is valid XML

Can you see why this would fail to parse?

```
{:timestamp=>"2015-09-12T16:02:44.194000-0400", 
:message=>"Trouble parsing xml with XmlSimple", 
:source=>"message", 
:value=>"<af type=\"nursery\" id=\"13710\" timestamp=\"Sep 12 16:02:43 2015\" intervalms=\"2891.467\">\n  <minimum requested_bytes=\"144\" />\n  <time exclusiveaccessms=\"0.062\" meanexclusiveaccessms=\"0.052\" threads=\"1\" lastthreadtid=\"0x0000000003790C00\" />\n  <refs soft=\"6060\" weak=\"14750\" phantom=\"3819\" dynamicSoftReferenceThreshold=\"30\" maxSoftReferenceThreshold=\"32\" />\n  <nursery freebytes=\"0\" totalbytes=\"324065280\" percent=\"0\" />\n  <tenured freebytes=\"2220643144\" totalbytes=\"6073352192\" percent=\"36\" >\n    <soa freebytes=\"2165983048\" totalbytes=\"6018692096\" percent=\"35\" />\n    <loa freebytes=\"54660096\" totalbytes=\"54660096\" percent=\"100\" />\n  </tenured>\n  <pending-finalizers finalizable=\"0\" reference=\"0\" classloader=\"0\" />\n  <gc type=\"scavenger\" id=\"13710\" totalid=\"13718\" intervalms=\"2891.699\">\n    <flipped objectcount=\"854555\" bytes=\"51632504\" />\n    <tenured objectcount=\"934\" bytes=\"68368\" />\n    <finalization objectsqueued=\"6\" />\n    <scavenger tiltratio=\"83\" />\n    <nursery freebytes=\"285158440\" totalbytes=\"337155072\" percent=\"84\" tenureage=\"14\" />\n    <tenured freebytes=\"2220422504\" totalbytes=\"6073352192\" percent=\"36\" >\n      <soa freebytes=\"2165762408\" totalbytes=\"6018692096\" percent=\"35\" />\n      <loa freebytes=\"54660096\" totalbytes=\"54660096\" percent=\"100\" />\n    </tenured>\n    <time totalms=\"34.007\" />\n  </gc>\n  <nursery freebytes=\"285092904\" totalbytes=\"337155072\" percent=\"84\" />\n  <tenured freebytes=\"2220422504\" totalbytes=\"6073352192\" percent=\"36\" >\n    <soa freebytes=\"2165762408\" totalbytes=\"6018692096\" percent=\"35\" />\n    <loa freebytes=\"54660096\" totalbytes=\"54660096\" percent=\"100\" />\n  </tenured>\n  <refs soft=\"6060\" weak=\"14750\" phantom=\"3819\" dynamicSoftReferenceThreshold=\"30\" maxSoftReferenceThreshold=\"32\" />\n  <pending-finalizers finalizable=\"6\" reference=\"0\" classloader=\"0\" />\n  <time totalms=\"34.256\" />\n</af>", :exception=>#<NoMethodError: undefined method `start_with?' for nil:NilClass>, :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.4-java/lib/logstash/event.rb:140:in `[]='", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-xml-1.0.0/lib/logstash/filters/xml.rb:129:in `filter'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.4-java/lib/logstash/filters/base.rb:163:in `multi_filter'", "org/jruby/RubyArray.java:1613:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.4-java/lib/logstash/filters/base.rb:160:in `multi_filter'", "(eval):765:in `cond_func_8'", "org/jruby/RubyArray.java:1613:in `each'", "(eval):762:in `cond_func_8'", "(eval):779:in `cond_func_7'", "org/jruby/RubyArray.java:1613:in `each'", "(eval):776:in `cond_func_7'", "(eval):667:in `filter_func'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.4-java/lib/logstash/pipeline.rb:219:in `filterworker'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.4-java/lib/logstash/pipeline.rb:157:in `start_filters'"], :level=>:warn}
```

```
<af type="nursery" id="13608" timestamp="Sep 12 15:59:55 2015" intervalms="6240.929">
  <minimum requested_bytes="80" />
  <time exclusiveaccessms="0.071" meanexclusiveaccessms="0.071" threads="0" lastthreadtid="0x0000000003786600" />
  <refs soft="6060" weak="14750" phantom="3817" dynamicSoftReferenceThreshold="30" maxSoftReferenceThreshold="32" />
  <nursery freebytes="0" totalbytes="322710528" percent="0" />
  <tenured freebytes="2528738016" totalbytes="6073352192" percent="41" >
    <soa freebytes="2474077920" totalbytes="6018692096" percent="41" />
    <loa freebytes="54660096" totalbytes="54660096" percent="100" />
  </tenured>
  <pending-finalizers finalizable="0" reference="0" classloader="0" />
  <gc type="scavenger" id="13608" totalid="13616" intervalms="6241.143">
    <flipped objectcount="699263" bytes="41617016" />
    <tenured objectcount="85357" bytes="5040808" />
    <finalization objectsqueued="6" />
    <scavenger tiltratio="81" />
    <nursery freebytes="285970608" totalbytes="327916544" percent="87" tenureage="13" />
    <tenured freebytes="2522668208" totalbytes="6073352192" percent="41" >
      <soa freebytes="2468008112" totalbytes="6018692096" percent="41" />
      <loa freebytes="54660096" totalbytes="54660096" percent="100" />
    </tenured>
    <time totalms="38.985" />
  </gc>
  <nursery freebytes="285905072" totalbytes="327916544" percent="87" />
  <tenured freebytes="2522668208" totalbytes="6073352192" percent="41" >
    <soa freebytes="2468008112" totalbytes="6018692096" percent="41" />
    <loa freebytes="54660096" totalbytes="54660096" percent="100" />
  </tenured>
  <refs soft="6060" weak="14750" phantom="3817" dynamicSoftReferenceThreshold="30" maxSoftReferenceThreshold="32" />
  <pending-finalizers finalizable="6" reference="0" classloader="0" />
  <time totalms="39.254" />
</af>
````

**Comment by wiibaa** [comment]
As the example xml is perfectly, I suppose you hit #9 because you "forgot" the `target` config.
I reproduced your error manually and confirmed that setting `target` fixes it!

@suyograo can you please close as a duplicate and see if someone can assign to review #26

**Comment by wiibaa** [comment]
Fixed by #26 

@suyograo can you close please
